src/libraries/Microsoft.Extensions.Configuration.Binder/gen/Helpers/Emitter/CoreBindingHelper.cs

```diff
@@ -429,10 +429,24 @@ private void EmitHelperMethods()
                     _emitBlankLineBeforeNextStatement = true;
                 }
 
+                bool enumTypeExists = false;
+
                 foreach (ParsableFromStringSpec type in _sourceGenSpec.PrimitivesForHelperGen)
                 {
                     EmitBlankLineIfRequired();
-                    EmitPrimitiveParseMethod(type);
+
+                    if (type.StringParsableTypeKind == StringParsableTypeKind.Enum)
```

Comment:
I think the parser should not include enum types in the `PrimitivesForHelperGen` list. The emitter should have minimal post-processing of the input data. The `SourceGenerationSpec` could have a boolean property indicating whether an enum was part of the input graph.